src/sentry/static/sentry/app/views/settings/projectDataPrivacy/projectDataPrivacyRulesPanel.tsx

```diff
@@ -0,0 +1,214 @@
+      setTimeout(async function() {
+        resolve(requestResponse);
+      }, 1000);
+    });
+
+    this.setState({
+      rules: result,
+      savedRules: result,
+      status: undefined,
+    });
+  };
+
+  formRef: React.RefObject<HTMLFormElement> = React.createRef();
+
+  handleAddRule = () => {
+    this.setState(prevState => ({
+      rules: [
+        ...prevState.rules,
+        {
+          id: prevState.rules.length + 1,
+          action: ACTION_TYPE.MASK,
+          data: DATA_TYPE.BANK_ACCOUNTS,
+          from: '',
+        },
+      ],
+    }));
+  };
+
+  handleDeleteRule = (ruleId: number) => () => {
+    this.setState(prevState => ({
+      rules: prevState.rules.filter(rule => rule.id !== ruleId),
+    }));
+  };
+
+  handleChange = (updatedRule: Rule) => {
+    this.setState(prevState => ({
+      ...prevState,
+      rules: prevState.rules.map(rule => {
+        if (rule.id === updatedRule.id) {
+          return updatedRule;
+        }
+        return rule;
+      }),
+    }));
+  };
+
+  handleSubmit = () => {
+    this.setState(
+      prevState => ({
+        status: 'success',
+        savedRules: prevState.rules,
+      }),
+      () => {
+        requestResponse = this.state.rules;
+      }
+    );
+  };
+
+  handleValidation = () => {
+    const {rules} = this.state;
+    const isAnyRuleFieldEmpty = rules.find(rule =>
+      Object.keys(rule).find(ruleKey => !rule[ruleKey])
+    );
+
+    const isFormValid = !isAnyRuleFieldEmpty;
+
+    if (isFormValid) {
+      this.formRef.current?.dispatchEvent(new Event('submit'));
+    } else {
+      this.setState({
+        status: 'error',
+      });
+    }
+  };
+
+  handleSaveForm = (event: React.MouseEvent) => {
+    event.stopPropagation();
+    this.handleValidation();
+  };
+
+  handleCancelForm = () => {
+    this.setState(prevState => ({
+      rules: prevState.savedRules,
+    }));
+  };
+
+  render() {
+    const {rules, status} = this.state;
+
+    if (status === 'loading') {
+      addMessage(t('Loading...'), status, {duration: 1000});
+    }
+
+    if (status === 'success') {
+      addMessage(t('Success'), status, {duration: 2000});
+    }
+
+    if (status === 'error') {
+      addMessage(t('An error occurred while saving the form'), status);
+    }
+
+    return (
+      <React.Fragment>
+        <Panel>
+          <PanelHeader>{t('Data Privacy Rules')}</PanelHeader>
+          <PanelBody>
+            <form
+              style={{marginBottom: 0}}
+              onSubmit={this.handleSubmit}
+              ref={this.formRef}
+            >
+              {rules.map(rule => (
+                <ProjectDataPrivacyRulesForm
+                  key={rule.id}
+                  onDelete={this.handleDeleteRule(rule.id)}
+                  onChange={this.handleChange}
+                  {...rule}
+                />
+              ))}
+            </form>
+            <PanelAction>
+              <StyledLink onClick={this.handleAddRule}>
```

Comment:
Since this is not navigating anywhere, can we make this a `<Button>`?

Reply:
I thought it would be better to use the link, because I would have to remove the ButtonLabel padding and style it a little ... I was thinking of creating a buttonLabelStyles prop to style the buttonLabel (didn't do it though) ... the button component is an interesting component to be refactored